pkg/skaffold/deploy/helm.go

```diff
@@ -161,7 +161,7 @@ func (h *HelmDeployer) deployRelease(ctx context.Context, out io.Writer, r lates
 	// Dependency builds should be skipped when trying to install a chart
 	// with local dependencies in the chart folder, e.g. the istio helm chart.
 	// This decision is left to the user.
-	if !r.SkipBuildDependencies {
```

Comment:
we should do renames only if it's really necessary - I feel like the benefit here is marginal - even though I understand where you're coming from. Do you mind reverting this?